docs/types/linter-message-v2.md

```diff
@@ -7,24 +7,31 @@ This document describes the type of Linter Message v2. It's supported in
 
 ```js
 type Message = {
-  // Automatically added for UI consumers, do not specify in a provider
+  // NOTE: DO NOT SPECIFY THESE IN PROVIDER
+  // Automatically added by base linter for UI consumers
   key: string,
   version: 2,
   linterName: string,
 
-  // From providers
+  // NOTE: These are given by providers
   location: {
     file: string,
     position: Range,
   },
-  source?: {
+  // ^ Location of the issue (aka where to highlight)
+  reference?: {
     file: string,
     position?: Point,
   },
+  // ^ Reference link to the issue, useful for jumping to classes etc
```

Comment:
How about this:
"Reference to a different location in the editor, useful for jumping to classes etc."